### PR TITLE
Add a way to give definitions of terms as tooltips #31

### LINKED ISSUE #31: Add a way to give definitions of terms as tool tips
e.g. when the reader hovers over a keyword, it pops up a tool tip containing the definition of the word

### PR DESCRIPTION
**Pull request by acjh**
Fixes #31
[Preview: Week 0](http://rawgit.com/acjh/website/31-add-way-definition-tooltips/contents/week.html?preview=0)

This is still a proof-of-concept, suggestions welcome :)

> A hyperlink in a `tooltip` is **not** meant to be clicked, so don't get frustrated over trying to click it.
> It's just there to demo HTML content. But yes, it works.

### CODE REVIEW

**Comment by damithc** [comment]
This is good, but perhaps we should also cater for cases where the term can be given in different ways?
One option: `<tooltip>Refactoring</tooltip>`, `<tooltip term='Refactoring'>Refactor</tooltip>`
Another option is to supply all aliases where we define the terms.

**Comment by acjh** [comment]
+1 for supplying all aliases where we define the terms, which would mean less clutter in the HTML :)
It's also easier to maintain (and to swop the 'key') by keeping track of all tooltip data in a single file.

File: scripts/tooltip.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Maps a string to a definition (in the form of a HTML string)
+var definitions = {
```
Comment (damithc) [inline scripts/tooltip.js]: `tooltips`? We don't have to limit this to definitions only.

**Comment by damithc** [comment]
The downside is you cannot have multiple definitions for the same keyword. Sometimes the same term can have different definitions depending on the context. Also there can be a simple definition and a longer definition. So the ideal is to be able to do both ways. But this is good enough for now. Adding the other method can be a separate issue.

**Comment by acjh** [comment]
Ready for review.

**Comment by damithc** [comment]
Can merge. Create an issue for the enhancement we discussed above?

**Comment by damithc** [comment]
@acjh 
Can you give an example of how to add a tooltip (including aiases)? I think the current code doesn't have any examples to follow. You can give the example here in this issue thread and I'll follow that to add a few tooltips.

**Comment by acjh** [comment]
Oh, I didn't want to pull the demo.
See [this commit](https://github.com/acjh/website/commit/d5c6524d3e2363418d25cc7149a0c6d73c6d1ad4).

**Comment by damithc** [comment]
Got it. 👍

**Pull request merged**